File: Baekjoon/2667_단지번호붙이기.py
from re import L
import sys
def input():
    return sys.stdin.readline().rstrip()

N = int(input())
lst = list(list(map(int,input())) for _ in range(N))
visited = list(list(0 for _ in range(N)) for _ in range(N))
answer = [0, []]

def dfs(y,x):
    global answer
    stk = [(y,x)]
    visited[y][x] = 1
    cnt = 0
    while stk:
        y,x = stk.pop()
        cnt += 1
        for dy, dx in [(-1,0), (1,0), (0,-1), (0,1)]:
            ny = y + dy
            nx = x + dx
            if 0 <= ny < N and 0 <= nx < N and lst[ny][nx] and not visited[ny][nx]:
                visited[ny][nx] = 1
                stk.append((ny,nx))
    else:
        answer[0] += 1
        answer[1].append(cnt)

for y in range(N):
    for x in range(N):
        if lst[y][x] and not visited[y][x]:
            dfs(y,x)
answer[1].sort()
print(answer[0])
for i in answer[1]:
    print(i)

File: Baekjoon/js_baekjoon/2667_단지번호붙이기.js
const fs = require('fs');
let input = fs.readFileSync('./input.txt').toString().trim().split('\n');
// let input = fs.readFileSync('./dev/stdin').toString().trim().split('\n');

const N = input[0]
const arr = input.slice(1).map((line) => Array.from(line))
const visited = Array.from({ length:N }, () => Array.from({ length:N }, () => false))

const dy = [-1, 0, 1, 0]
const dx = [0, 1, 0, -1]

const answer = [0,[]]

function dfs(y,x) {
  const stk_y = [y]
  const stk_x = [x]
  let cnt = 0
  visited[y][x] = true
  while (stk_x.length > 0) {
    y = stk_y.pop()
    x = stk_x.pop()
    cnt += 1
    for (let i = 0; i < 4; i++) {
      ny = y + dy[i]
      nx = x + dx[i]
      if (0 <= ny && ny < N && 0 <= nx && nx < N && !visited[ny][nx] && arr[ny][nx] == 1) {
        stk_y.push(ny)
        stk_x.push(nx)
        visited[ny][nx] = true
      }
    }
  }
  answer[0] += 1
  answer[1].push(cnt)
}

for (let y = 0; y < N; y++) {
  for (let x = 0; x < N; x++) {
    if (!visited[y][x] && arr[y][x] == 1) {
      dfs(y,x)
    }
  }
}

answer[1].sort((a,b) => (a-b))

console.log(answer[0])
for (const element of answer[1]) {
  console.log(element)
}